lib/screens/login_screen.dart

```diff
@@ -24,80 +26,181 @@ class LoginScreen extends HookConsumerWidget {
+                                    .errorConnectingToServer,
+                              ),
+                              content: e.response?.statusCode == null
+                                  ? Text(e.toString())
+                                  : Text(formatHttpErrorCode(e.response)),
+                            );
+                            return;
+                          } catch (e) {
+                            if (!context.mounted) return;
+                            await showInfoDialog(
+                              context,
+                              Text(AppLocalizations.of(context)!
+                                  .errorConnectingToServer),
+                              content: Text(e.toString()),
+                            );
+                            return;
                           }
-                        } catch (e) {
-                          // TODO: show error message to user
-                          //print(e);
-                        }
-                      },
-                      child: Text(AppLocalizations.of(context)!.login),
+                        },
+                        child: Text(AppLocalizations.of(context)!.login),
+                      ),
                     ),
                   ),
-                ),
-                kIsWeb
-                    ? Text(AppLocalizations.of(context)!.webDemoNote)
-                    : const SizedBox(),
-              ],
+                  kIsWeb
+                      ? Text(AppLocalizations.of(context)!.webDemoNote)
+                      : const SizedBox(),
+                ],
+              ),
             ),
           ),
         ),
       )),
     );
   }
+
+  Future<void> showInfoDialog(
+    BuildContext context,
+    Widget title, {
+    Widget? content,
+  }) async {
+    await showDialog(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: title,
+        content: content,
+        actions: [
+          ElevatedButton(
+            onPressed: () {
+              Navigator.pop(context);
+            },
+            child: const Text('Ok'),
+          )
+        ],
+      ),
+    );
+  }
+
+  String formatMissingFields(BuildContext context, String username,
+      String password, String serverAddress) {
+    var missingFields = '';
+    if (username.isEmpty) {
+      missingFields += '${AppLocalizations.of(context)!.emptyUsername}\n\n';
+    }
+    if (password.isEmpty) {
+      missingFields += '${AppLocalizations.of(context)!.emptyPassword}\n\n';
+    }
+    if (serverAddress.isEmpty) {
+      missingFields += '${AppLocalizations.of(context)!.emptyAddress}\n\n';
+    }
+
+    return missingFields;
+  }
+
+  String formatHttpErrorCode(Response? resp) {
+    var message = '';
+    switch (resp!.statusCode) {
+      case 400:
+        message =
+            'Something went wrong while making the request, this is probably a issue within Jellyflix, please open a github issue';
```

Comment:
These need to reflect more standard representations of the code status.  401 is fine because this is a totally standard way of using it in jellyland, but the other two are a bit misleading. just go with "400 - Bad Request" or "403 - Forbidden"

Reply:
I don't mind leaving them out, but here's my reasoning

400 - can only happen if the client library (tentacle) messes up a request or the server doesn't recognize the path for whatever reason either way its a bug within the app

403 - I guess is debatable but this only occurs if the server has banned your device IP or ratelimit I guess either way still relevant message.

My assumptions could be wrong so feel free to correct me.

Reply:
imho it depends where in the flow we are making this determination. if we have already validated that we are talking to a jellyfin server and not something else in error, I think this 400 is ok.  the 403 is a problem though because lots of different reverse proxy stuff could cause this.

Reply:
tbh i have problems with the 400 too because of the potential issues with proxies.  if there is a 400 or 403, it would be really nice to give the user as much info about it as possible instead of generalized lines.  what do you think about using the response content itself in these cases?
@jdk-21 do you have an opinion on this one?

Reply:
ah i hadn't considered proxies, one solution could be to validate that we are talking to jellyfin first, based on this [auth logic](https://github.com/jellyflix-app/jellyflix/pull/124#discussion_r1740206079), if we get 400ed before that we show a normal exception

Reply:
i have reservations about overcasting the 400 or the 403.  im not an expert in this area but my instinct is usually to avoid taking any particular networking configuration for granted.  401 is an exception because its the only http code that can only ever mean one thing -- no matter what, no matter where it happened, the credentials were wrong.  so in that case its fine to say something like "the credentials were wrong."
In every other case I am super skeptical of overcasting the real error, but I'm open to advice.  If I have time I will take a look at how other clients do it.

Reply:
yeah makes sense. I could change the messages to be very general, instead of jellyfin specific 

eg for 400 something like 

_the server could not understand the request, if you are using proxies, please check them first if the issue still persists, ask in discord._

Reply:
truth be told I don't see the wisdom in this function at all.  this is a bit backwards in general because 1) you are covering over information about the real error that might be useful to the user or troubleshooter and then 2) you funnel these people into the support channels with less info than they might have had. this has negative impacts both on the UX and the support burden.
why isn't it better to unconditionally use Text(e.toString())

Reply:
To be clear, the function does not hide the HTTP error, it adds this message on top of the HTTP response (see image in the PR)

e.tostring will use dios inbuilt error response, which is a long string you can see here.

![sev](https://github.com/user-attachments/assets/85d65f49-c12c-47f8-bd9c-53aa3c876365)

Reply:
ah, well I take it back, I see the wisdom in that.  how do you feel about the ux/support burden element?

Reply:
If we can workshop the messages, and add like a FAQ section, I feel like it shouldn't be too bad.

Reply:
this is the final message that gets displayed, let me know if you want additional info included

```dart
return '$message\n\n'
            'Http Code: ${resp.statusCode ?? 'Unknown'}\n\n'
            'Http Response: ${resp.statusMessage ?? 'Unknown'}\n\n'
        .trim();
```